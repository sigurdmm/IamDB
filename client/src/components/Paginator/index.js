import Pagination from 'react-js-pagination';
import PropTypes from 'prop-types';

import './index.less';

const pageToOffset = (page, limit) => (page - 1) * limit;
const offsetToPage = (offset, limit) => (offset / limit) + 1;

const Paginator = (props) => {
  const {
    offset,
    limit,
    total,
    onPagination,
  } = props;
  console.info(`offset: ${offset}, limit: ${limit}, total: ${total}`);

  const pageCount = 5;

  return <Pagination activePage={offsetToPage(offset, limit)}
                     totalItemsCount={total}
                     itemsCountPerPage={limit}
                     pageRangeDisplayed={pageCount}
                     onChange={page => onPagination(pageToOffset(page, limit))}/>;
};

Paginator.propTypes = {
  offset: PropTypes.number.isRequired,
  limit: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  onPagination: PropTypes.func.isRequired,
};

export default Paginator;
