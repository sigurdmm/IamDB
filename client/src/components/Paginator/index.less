
.pagination {
  background-color: #fff;
  display: inline-block;
  padding-left: 0;
  margin: 20px 0;
  border-radius: 4px;

  text-align: center;

  > li {
    box-sizing: border-box;

    display: inline;

    > a {
      box-sizing: border-box;
      position: relative;
      float: left;
      padding: 6px 12px;
      line-height: 1.42857143;
      text-decoration: none;
      color: #337ab7;
      background-color: #fff;
      border: 1px solid #ddd;
      margin-left: -1px;
    }
  }

  .disabled > a,
  .disabled > a:focus,
  .disabled > a:hover {
    color: #777;
    background-color: #fff;
    border-color: #ddd;
    cursor: not-allowed;
  }

  .active > a,
  .active > a:focus,
  .active > a:hover {
    z-index: 3;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
    cursor: default;
  }
}
