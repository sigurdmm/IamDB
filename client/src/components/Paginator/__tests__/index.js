import { render } from 'enzyme';

import Paginator from '../index';

describe('<Paginator/>', () => {
  it('should render properly', () => {
    const wrapper = render(<Paginator
      offset={0}
      limit={50}
      total={100}
      onPagination={() => null}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
