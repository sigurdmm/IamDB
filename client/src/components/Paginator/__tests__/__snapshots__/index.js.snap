// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator/> should render properly 1`] = `
<ul
  class="pagination"
>
  <li
    class="disabled"
  >
    <a
      class=""
      href="#"
    >
      first
    </a>
  </li>
  <li
    class="disabled"
  >
    <a
      class=""
      href="#"
    >
      prev
    </a>
  </li>
  <li
    class="active"
  >
    <a
      class="undefined"
      href="#"
    >
      1
    </a>
  </li>
  <li
    class=""
  >
    <a
      class=""
      href="#"
    >
      2
    </a>
  </li>
  <li
    class=""
  >
    <a
      class=""
      href="#"
    >
      next
    </a>
  </li>
  <li
    class=""
  >
    <a
      class=""
      href="#"
    >
      last
    </a>
  </li>
</ul>
`;
